github/resource_github_actions_secret.go

```diff
@@ -0,0 +1,157 @@
+package github
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"github.com/google/go-github/v29/github"
+	"github.com/hashicorp/terraform/helper/schema"
+	"golang.org/x/crypto/nacl/box"
+	"log"
+)
+
+func resourceGithubActionsSecret() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubActionsSecretCreateOrUpdate,
+		Read:   resourceGithubActionsSecretRead,
+		Update: resourceGithubActionsSecretCreateOrUpdate,
+		Delete: resourceGithubActionsSecretDelete,
+
+		Schema: map[string]*schema.Schema{
+			"owner": {
```

Comment:
Why is the `owner` argument required?  Given you use [`checkOrganization`](https://github.com/terraform-providers/terraform-provider-github/blob/b838aed6391ac450740f023e4a714d6cd449aad3/github/util.go#L16) func, that checks that organization is configured in provider settings, you could get the value from provider config via `meta`. As an example, take a look at [repository webhook](https://github.com/terraform-providers/terraform-provider-github/blob/8dc722aeae3d86b082447e22f34ac69539bc277c/github/resource_github_repository_webhook.go#L102) code.

Reply:
I was unaware that I could pull the organisation from the `meta` to be honest!
Thinking about this, would there be any case that you would want to manage a secret that wasn't part of the organisation in the provider? I think it might be possible in the current implementation but should not be encouraged - forcing users to have an individual provider per organisation seems more appropriate?

Reply:
Yes, same ideology is used when managing other repository resources, webhooks is one example.